internal/fleet/agent_policy/models.go

```diff
@@ -851,6 +899,213 @@ func mergeAgentFeature(existing []apiAgentFeature, newFeature *apiAgentFeature)
 	return &result
 }
 
+// advancedSettingsAttrTypes returns attribute types for advanced_settings pulled from the schema
+func advancedSettingsAttrTypes() map[string]attr.Type {
+	return getSchema().Attributes["advanced_settings"].GetType().(attr.TypeWithAttributeTypes).AttributeTypes()
+}
+
+// populateAdvancedSettingsFromAPI populates the advanced settings from API response
+func (model *agentPolicyModel) populateAdvancedSettingsFromAPI(ctx context.Context, data *kbapi.AgentPolicy) diag.Diagnostics {
```

Comment:
This file is getting pretty big. WDYT about pulling these out to `models_advanced_settings.go` etc?